Userland/Libraries/LibJS/Runtime/Temporal/AbstractOperations.cpp

```diff
@@ -800,8 +810,8 @@ Optional<TemporalTimeZone> parse_temporal_time_zone_string(GlobalObject& global_
 
     // 4. If z is not undefined, then
     if (z_part.has_value()) {
-        // a. Return the Record { [[Z]]: "Z", [[OffsetString]]: "+00:00", [[Name]]: undefined }.
-        return TemporalTimeZone { .z = true, .offset = "+00:00", .name = {} };
+        // a. Return the Record { [[Z]]: true, [[OffsetString]]: undefined, [[Name]]: name }.
+        return TemporalTimeZone { .z = true, .offset = {}, .name = name_part.has_value() ? String { *name_part } : String {} };
```

Comment:
how about:
```c++
return TemporalTimeZone { .z = true, .offset = {}, .name = String { name_part.value_or({}) }};
```
I think it looks a bit nicer, but up to you

Reply:
I don't think that's functionally equivalent, `Optional<T>::value_or(T&)` still returns a `T`, so in this case a null `StringView`, and I guess a null string - we want an empty optional...

Reply:
> we want an empty optional

oh no here we don't. good catch!

Reply:
> we want an empty optional

... and I messed that part up :^)

Reply:
Changed `String {}` to `Optional<String> {}`. Still looks gross, but that's how `Optional<StringView>` -> `Optional<String>` conversion works, I suppose.